pkg/controllers/customcluster_controller.go

```diff
@@ -201,12 +195,27 @@ func (r *CustomClusterController) Reconcile(ctx context.Context, req ctrl.Reques
 			return ctrl.Result{}, nil
 		}
 		log.Error(err, "failed to find customMachine", "customMachine", customMachinekey)
-		return ctrl.Result{RequeueAfter: RequeueAfter}, err
+		return ctrl.Result{}, err
+	}
+
+	// Fetch the KubeadmControlPlane instance.
+	kcpKey := client.ObjectKey{
+		Namespace: cluster.Spec.ControlPlaneRef.Namespace,
+		Name:      cluster.Spec.ControlPlaneRef.Name,
+	}
+	kcp := &controlplanev1.KubeadmControlPlane{}
+	if err := r.Client.Get(ctx, kcpKey, kcp); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("kcp does not exist", "kcp", kcpKey)
+			return ctrl.Result{}, nil
+		}
+		log.Error(err, "failed to get kcp", "kcp", kcpKey)
+		return ctrl.Result{}, err
```

Comment:
if there's a error(mostly like a network issue), should add RequeueAfter?

Reply:
IIRC, the controller runtime does not respect requeueAfter at all when there is an err

```
	result, err := c.Reconcile(ctx, req)
	switch {
	case err != nil:
		c.Queue.AddRateLimited(req)
		ctrlmetrics.ReconcileErrors.WithLabelValues(c.Name).Inc()
		ctrlmetrics.ReconcileTotal.WithLabelValues(c.Name, labelError).Inc()
		log.Error(err, "Reconciler error")
```